### PR TITLE
Delete icon button

### PR DESCRIPTION
**Pull request by spireken**

### CODE REVIEW

**Review by shaobinspire — CHANGES_REQUESTED** [review]
There is a border when the button gets focus, but I don't see it in the specs of the DeleteIconButton.

File: Applications/Spire/Source/Ui/Button.cpp [hunk 1]
```diff
@@ -101,6 +101,20 @@ Button* Spire::make_icon_button(QImage icon, QWidget* parent) {
...
 }
 
+Button* Spire::make_delete_icon_button(QWidget* parent) {
+  auto button = make_icon_button(imageFromSvg(":/Icons/delete.svg",
+    scale(16, 16)), parent);
+  auto style = get_style(*button);
+  style.get(Body()).
+    set(BackgroundColor(QColor::fromRgb(0, 0, 0, 0))).
+    set(Fill(QColor::fromRgb(0xC8, 0xC8, 0xC8)));
+  style.get(Focus() / Body() || Hover() / Body()).
```
Comment (kranar) [inline Applications/Spire/Source/Ui/Button.cpp:111]: Factor this out ```(Focus() || Hover()) / Body()```

**Comment by spireken** [comment]
> There is a border when the button gets focus, but I don't see it in the specs of the DeleteIconButton.

The DeleteIconButton is an IconButton so I believe it inherits the IconButton style.

**Review by shaobinspire — APPROVED** [review]

**Comment by kranar** [comment]
I believe work still needs to be done on this after some changes to icon, resubmit it later.

**Pull request closed without merging**